pkg/loop/internal/test/test.go

```diff
@@ -49,10 +79,11 @@ var (
 		ID:     chainID,
 		Config: ConfigTOML,
 	}
-	chainID            = "chain-id"
-	configDigest       = libocr.ConfigDigest([32]byte{2: 10, 12: 16})
-	configDigestPrefix = libocr.ConfigDigestPrefix(99)
-	contractConfig     = libocr.ContractConfig{
+	chainID                   = "chain-id"
+	configDigest              = libocr.ConfigDigest([32]byte{2: 10, 12: 16})
+	configDigestPrefix        = libocr.ConfigDigestPrefix(99)
+	latestTransmissionDetails = LatestTransmissionDetails{ConfigDigest: marshalableConfigDigest(configDigest), Epoch: epoch, Round: round, Timestamp: latestTimestamp}
```

Comment:
IMHO a struct is not worthwhile here. It introduces extra complexity not relevant to the test. A regular map is sufficient to exercise the serialization and can be compared directly without extra steps:
```suggestion
	latestTransmissionDetails = map[string]interface{}{...}
```

Reply:
I did try that, but ended up reverting it back because it causes the test to fail.  (I saved it on a branch in case you feel strongly about it and want us to try again to get something along those lines to work, but hear me out first...)

The error message it fails with is this:
```
2023-10-31T14:20:43.715-0700	ERROR	MedianService.PluginMedianClient.MedianPluginFactory	Client refresh attempt failed	{"err": "rpc error: code = Unknown desc = GetLatestValue: expected map[ConfigDigest:[0 0 10 0 0 0 0 0 0 0 0 0 16 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0] Epoch:88 LatestAnswer:<nil> Round:74 Timestamp:2009-02-13T23:31:30.987654321Z] but got map[ConfigDigest:00000a0000000000000000001000000000000000000000000000000000000000 Epoch:88 Round:74 Timestamp:2009-02-13 23:31:30.987654321 +0000 UTC]"}
```

The reason it fails is because there is not enough type information in `map[string]any` for it to know that the hex string is supposed to be converted back into bytes rather than remain a string.  If you Marshal `[]byte` into json then you get a hex string.  If you unmarshal it back into `[]byte` it works, but trying to unmarshal back into `any` does not work.  So a map will necessarily fail in production for the same reason.  I don't think we need to support arbitrary maps though--it seems fine if we only support passing a `struct` with the proper type information included for deserializing.

Reply:
IMHO converting from struct to json and back to struct is an important part of the basic serialization/deserialization process, so it seems natural to me that it be included in basic tests of serialization across gRPC.

Reply:
> for it to know that the hex string is supposed to be converted back into bytes rather than remain a string

It shouldn't be trying to do that. That is out of scope for this test. I am proposing that we use _simpler_ types which do not require special cases for encoding/decoding.

> IMHO converting from struct to json and back to struct is an important part of the basic serialization/deserialization process,

It is an entirely separate layer of serialization that _must_ be covered in other tests, because these ones are stubs by design, focused only on the GRPC protobuf serialization.

Regardless, the server side code will _never_ have access to the struct types, so it doesn't make sense to write test code that uses it this way.

Reply:
> Regardless, the server side code will never have access to the struct types, so it doesn't make sense to write test code that uses it this way.

Maybe I'm just confused about this, but I'm pretty sure the tests run on the client side calling the client method... so shouldn't they imitate how the client method will actually be called? And receive a struct back for a return val as the client would actually be receiving?  I'm not sure why the server side is relevant here.

Reply:
I could replace this one with a simpler test that just uses very basic types, but we definitely need this test so... we need some place to move it and I"m not sure where else it would go.  Did you have any place in mind?

Reply:
These static test types are strictly for exercising the GRPC protobuf serialization between clients and servers. They are only able to do one thing at a time, so we will never be able to achieve full coverage with these tests alone. We must have additional coverage for your unique logic to ensure that structs, maps, and arbitrary json.Marshal/Unmarshal methods each function as expected, likely with real code instead of this test stub.

Reply:
ok... I think I see where it would go, after looking over the tests added in the recently merged branches--I'll move it to there